http_service/bugbug_http/app.py

```diff
@@ -452,6 +500,81 @@ def batch_prediction(model_name):
     return jsonify({"bugs": data}), status_code
 
 
+@application.route("/push/<path:branch>/<rev>/schedules")
+@cross_origin()
+def push_schedules(branch, rev):
+    """
+    ---
+    get:
+      description: Determine which tests and tasks a push should schedule.
+      summary: Get which tests and tasks to schedule.
+      parameters:
+      - name: branch
+        in: path
+        schema:
+          BranchName
+      - name: rev
+        in: path
+        schema:
+          type: str
+          example: 76383a875678
+      responses:
+        200:
+          description: A dict of tests and tasks to schedule.
+          content:
+            application/json:
+              schema: Schedules
+        202:
+          description: Request is still being processed.
+          content:
+            application/json:
+              schema: NotAvailableYet
+        401:
+          description: API key is missing
+          content:
+            application/json:
+              schema: UnauthorizedError
+        404:
+          description: Push not found.
+          content:
+            application/json:
+              schema: PushNotFoundError
+    """
+    headers = request.headers
+    redis_conn.ping()
+
+    auth = headers.get(API_TOKEN)
+
+    if not auth:
+        return jsonify(UnauthorizedError().dump({})), 401
+    else:
+        LOGGER.info("Request with API TOKEN %r", auth)
+
+    # Support the string 'autoland' for convenience.
+    if branch == "autoland":
+        branch = "integration/autoland"
+
+    url = f"https://hg.mozilla.org/{branch}/json-automationrelevance/{rev}"
+    r = requests.get(url)
```

Comment:
We might want to move this to the worker side to keep the web dyno always quickly responsive, as the request could be slow.